src/CodeGen_LLVM.h

```diff
@@ -106,11 +106,12 @@ class CodeGen_LLVM : public IRVisitor {
     virtual void end_func(const std::vector<LoweredArgument> &args);
     // @}
 
-    /** What should be passed as -mcpu, -mattrs, and related for
-     * compilation. The architecture-specific code generator should
-     * define these. */
+    /** What should be passed as -mcpu (warning: implies attrs!), -mattrs,
+     *  and related for compilation. The architecture-specific code generator
+     *  should define these. */
     // @{
-    virtual std::string mcpu() const = 0;
+    virtual std::string mcpu_target() const = 0;
```

Comment:
Possibly-dumb question: would it make sense to refactor all of these into a single virtual method returning a struct? How common is it to override only one of target/tune/attrs without affecting the others?

Reply:
No idea, you are asking wrong person :)
The obvious problem is that when a new field is added,
instead of loudly failing to compile at every place that needs to be updated,
you get some default garbage. So i wouldn't do that.